apps/browser-proxy/src/pg-dump-middleware/pg-dump-middleware.ts

```diff
@@ -0,0 +1,83 @@
+import ExpiryMap from 'expiry-map'
+import type { ClientParameters } from 'pg-gateway'
+import { isGetExtensionsQuery, patchGetExtensionsResult } from './get-extensions-query.ts'
+import {
+  isGetExtensionMembershipQuery,
+  patchGetExtensionMembershipResult,
+} from './get-extension-membership-query.ts'
+import { FIRST_NORMAL_OID } from './constants.ts'
+
+type ConnectionId = string
+
+const state = new ExpiryMap<ConnectionId, State>(1000 * 60 * 5)
```

Comment:
Worth adding a comment explaining `ExpiryMap`? Why did we need this?

Reply:
I refactored the middleware so we don't need ExpiryMap. We are now cleaning the middleware state for a given connectionId when its TCP socket is closed.